With gcc it's ok, just the one warning about `invalidatefilter`. But with clang I get really many lines like
```
[ 43%] Building CXX object src/CMakeFiles/fm-qt6.dir/core/fileinfojob.cpp.o
In file included from /tmp/makepkg/libfm-qt-git/src/libfm-qt/src/core/filemonitor.cpp:1:
In file included from /tmp/makepkg/libfm-qt-git/src/libfm-qt/src/core/filemonitor.h:6:
In file included from /tmp/makepkg/libfm-qt-git/src/libfm-qt/src/core/gioptrs.h:5:
In file included from /usr/include/glib-2.0/glib.h:34:
In file included from /usr/include/glib-2.0/glib/gasyncqueue.h:34:
In file included from /usr/include/glib-2.0/glib/gthread.h:36:
/usr/include/glib-2.0/glib/gutils.h:378:7: warning: '__COUNTER__' is a C2y extension [-Wc2y-extensions]
  378 |   if (G_UNLIKELY (nth_bit < -1))
      |       ^
/usr/include/glib-2.0/glib/gmacros.h:1285:45: note: expanded from macro 'G_UNLIKELY'                              
 1285 | #define G_UNLIKELY(expr) (__builtin_expect (_G_BOOLEAN_EXPR(expr), 0))
      |                                             ^
/usr/include/glib-2.0/glib/gmacros.h:1283:53: note: expanded from macro '_G_BOOLEAN_EXPR'                         
 1283 | #define _G_BOOLEAN_EXPR(expr) _G_BOOLEAN_EXPR_IMPL (__COUNTER__, expr)
      |                                                     ^
In file included from /tmp/makepkg/libfm-qt-git/src/libfm-qt/src/core/filemonitor.cpp:1:                          
In file included from /tmp/makepkg/libfm-qt-git/src/libfm-qt/src/core/filemonitor.h:6:
In file included from /tmp/makepkg/libfm-qt-git/src/libfm-qt/src/core/gioptrs.h:5:
In file included from /usr/include/glib-2.0/glib.h:34:
In file included from /usr/include/glib-2.0/glib/gasyncqueue.h:34:
In file included from /usr/include/glib-2.0/glib/gthread.h:36:
/usr/include/glib-2.0/glib/gutils.h:393:22: warning: '__COUNTER__' is a C2y extension [-Wc2y-extensions]
  393 |   if (nth_bit < 0 || G_UNLIKELY (nth_bit > GLIB_SIZEOF_LONG * 8))
      |                      ^
/usr/include/glib-2.0/glib/gmacros.h:1285:45: note: expanded from macro 'G_UNLIKELY'                              
 1285 | #define G_UNLIKELY(expr) (__builtin_expect (_G_BOOLEAN_EXPR(expr), 0))
      |                                             ^
/usr/include/glib-2.0/glib/gmacros.h:1283:53: note: expanded from macro '_G_BOOLEAN_EXPR'                         
 1283 | #define _G_BOOLEAN_EXPR(expr) _G_BOOLEAN_EXPR_IMPL (__COUNTER__, expr)
      |                                                     ^
In file included from /tmp/makepkg/libfm-qt-git/src/libfm-qt/src/core/filemonitor.cpp:1:                          
In file included from /tmp/makepkg/libfm-qt-git/src/libfm-qt/src/core/filemonitor.h:6:
In file included from /tmp/makepkg/libfm-qt-git/src/libfm-qt/src/core/gioptrs.h:5:
In file included from /usr/include/glib-2.0/glib.h:34:
In file included from /usr/include/glib-2.0/glib/gasyncqueue.h:34:
In file included from /usr/include/glib-2.0/glib/gthread.h:36:
/usr/include/glib-2.0/glib/gutils.h:408:10: warning: '__COUNTER__' is a C2y extension [-Wc2y-extensions]
  408 |   return G_LIKELY (number) ?
      |          ^
/usr/include/glib-2.0/glib/gmacros.h:1284:43: note: expanded from macro 'G_LIKELY'                                
 1284 | #define G_LIKELY(expr) (__builtin_expect (_G_BOOLEAN_EXPR(expr), 1))
      |                                           ^
/usr/include/glib-2.0/glib/gmacros.h:1283:53: note: expanded from macro '_G_BOOLEAN_EXPR'                         
 1283 | #define _G_BOOLEAN_EXPR(expr) _G_BOOLEAN_EXPR_IMPL (__COUNTER__, expr)
      |                                                     ^
In file included from /tmp/makepkg/libfm-qt-git/src/libfm-qt/src/core/filemonitor.cpp:1:                          
In file included from /tmp/makepkg/libfm-qt-git/src/libfm-qt/src/core/filemonitor.h:6:
In file included from /tmp/makepkg/libfm-qt-git/src/libfm-qt/src/core/gioptrs.h:5:
In file included from /usr/include/glib-2.0/glib.h:56:
In file included from /usr/include/glib-2.0/glib/giochannel.h:36:
/usr/include/glib-2.0/glib/gstring.h:204:7: warning: '__COUNTER__' is a C2y extension [-Wc2y-extensions]
  204 |   if (G_LIKELY (gstring != NULL &&
      |       ^
/usr/include/glib-2.0/glib/gmacros.h:1284:43: note: expanded from macro 'G_LIKELY'                                
 1284 | #define G_LIKELY(expr) (__builtin_expect (_G_BOOLEAN_EXPR(expr), 1))
      |                                           ^
/usr/include/glib-2.0/glib/gmacros.h:1283:53: note: expanded from macro '_G_BOOLEAN_EXPR'                         
 1283 | #define _G_BOOLEAN_EXPR(expr) _G_BOOLEAN_EXPR_IMPL (__COUNTER__, expr)
      |                                                     ^
In file included from /tmp/makepkg/libfm-qt-git/src/libfm-qt/src/core/filemonitor.cpp:1:                          
In file included from /tmp/makepkg/libfm-qt-git/src/libfm-qt/src/core/filemonitor.h:6:
In file included from /tmp/makepkg/libfm-qt-git/src/libfm-qt/src/core/gioptrs.h:5:
In file included from /usr/include/glib-2.0/glib.h:56:
In file included from /usr/include/glib-2.0/glib/giochannel.h:36:
/usr/include/glib-2.0/glib/gstring.h:226:6: warning: '__COUNTER__' is a C2y extension [-Wc2y-extensions]
  226 |   if G_UNLIKELY (gstring == NULL)
      |      ^
/usr/include/glib-2.0/glib/gmacros.h:1285:45: note: expanded from macro 'G_UNLIKELY'                              
 1285 | #define G_UNLIKELY(expr) (__builtin_expect (_G_BOOLEAN_EXPR(expr), 0))
      |                                             ^
/usr/include/glib-2.0/glib/gmacros.h:1283:53: note: expanded from macro '_G_BOOLEAN_EXPR'                         
 1283 | #define _G_BOOLEAN_EXPR(expr) _G_BOOLEAN_EXPR_IMPL (__COUNTER__, expr)
      |                                                     ^
In file included from /tmp/makepkg/libfm-qt-git/src/libfm-qt/src/core/filemonitor.cpp:1:                          
In file included from /tmp/makepkg/libfm-qt-git/src/libfm-qt/src/core/filemonitor.h:6:
In file included from /tmp/makepkg/libfm-qt-git/src/libfm-qt/src/core/gioptrs.h:5:
In file included from /usr/include/glib-2.0/glib.h:56:
In file included from /usr/include/glib-2.0/glib/giochannel.h:36:
/usr/include/glib-2.0/glib/gstring.h:229:6: warning: '__COUNTER__' is a C2y extension [-Wc2y-extensions]
  229 |   if
  ....
  ```
  `lxqt-archiver is good with both gcc and clang.

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,5 +1,7 @@
 set(CMAKE_C_EXTENSIONS ON)
 
+add_compile_options("$<$<CXX_COMPILER_ID:Clang>:-Wno-c2y-extensions>")
+
 set(libfm_core_SRCS
     # gio gvfs implementations
     core/vfs/fm-file.c

